### PR TITLE
Refactors build to use a parent pom

### PR DESCRIPTION
**Pull request by adriancole**
This simplifies the build a lot by using a parent pom (as suggested by @trustin)

This simplifies further by moving some configuration management to `build_image`.

### CODE REVIEW

**Comment by adriancole** [comment]
Note: I initially planned to use properties files, but https://github.com/mojohaus/properties-maven-plugin/issues/26 thwarted me. This centralizes all the build config into maven pom rather than splitting into multiple locations.

**Review by anuraaga — APPROVED** [review]

File: docker/RATIONALE.md [hunk 1]
```diff
@@ -1,12 +1,8 @@
...
-layer. We need to know the target stage before we create it, so that we can
-build the right contents. There's no known way to read the value of the
-`--target` parameter. Hence, we duplicate it as a build argument.
-
-One impact of doing this is `DOCKER_BUILDKIT=1` or similar is required to skip
-unused phases present in the process of building the image.
+`build_image` is a script used to build Docker images per project. We need
+some details from the pom file for build arguments. Rather than require
+installing an XML parser or invoking Maven, this uses `sed` which is commonly
+present on developer laptions and CI nodes.
```
Comment (adriancole) [inline docker/RATIONALE.md:8]:
```suggestion
present on developer laptops and CI nodes.
```

**Comment by adriancole** [comment]
fingers crossed syntax is correct for both dockerhub and GH actions now :D

sorry about the thrash, I should have pushed a branch before making a PR out of it.

**Comment by adriancole** [comment]
finally green!

**Pull request merged**